HelpSource/Reference/Syntax-Shortcuts.schelp

```diff
@@ -53,66 +70,123 @@ Thing { var x;
     x { ^x }
     x_ { arg z; x = z; }
 }
-:: || code:: Thing { var <>x; } ::
+:: || code::
+Thing { var <>x; }
+::
 ::
 
-subsection:: calling an instance variable setter method
+
+section:: Sending messages, calling functions, and instantiating objects
+
+subsection:: functional and receiver notation equivalence
 table::
 ## instead of writing: || you can write:
-## code:: p.x_(y) :: || code:: p.x = y :: or code:: x(p) = y ::
+## code:: f(x, y) :: || code:: x.f(y) ::
+## code:: f(g(x)) :: || code:: x.g.f ::
 ::
 
-subsection:: use a selector as binary operator
+subsection:: calling the 'value' method
 table::
 ## instead of writing: || you can write:
-## code:: min(x, y) :: || code:: x min: y ::
+## code:: f.value(x) :: || code:: f.(x) ::
 ::
 
 subsection:: instantiate object
 table::
 ## instead of writing: || you can write:
-## code:: Point.new(3, 4); :: || code:: Point(3, 4) ::
+## code:: Point.new(3, 4) :: || code:: Point(3, 4) ::
 ::
 
-subsection:: moving blocks out of argument lists
+subsection:: calling an instance variable setter method
 table::
 ## instead of writing: || you can write:
-## code:: if (x<3, {\abc}, {\def}); :: || code:: if (x<3) {\abc} {\def} ::
-## code:: z.do({|x| x.play }); :: || code:: z.do {|x| x.play }; ::
-## code:: while({ a < b },{ a = a * 2 }); :: || code:: while { a < b } { a = a * 2 }; ::
+## code:: p.x_(y) :: || code:: p.x = y :: or code:: x(p) = y ::
 ::
 
-subsection:: shorter argument lists
+subsection:: calling performList
 table::
 ## instead of writing: || you can write:
-## code:: { arg x; x < 2 } :: || code:: {|x| x < 2 } ::
-## code:: { arg x = 123; x < 2 } :: || code:: {|x = 123| x < 2 } ::
-## code:: { arg x = 10.rand; x < 2 } :: || code:: {|x = (10.rand)| x < 2 } :: or code:: {|x(10.rand)| x < 2 } ::
-::
-note::
-When using the new code::||:: syntax, the default value needs to be enclosed in parenthesis if it's not a literal.
+## code:: object.performList(\method, a, b, array) :: || code:: object.method(a, b, *array) ::
 ::
 
-subsection:: calling the 'value' method
+
+
+subsection:: moving blocks out of argument lists (trailing-block arguments)
 table::
 ## instead of writing: || you can write:
-## code:: f.value(x) :: || code:: f.(x) ::
+## code:: if (x<3, {\abc}, {\def}) :: || code:: if (x<3) {\abc} {\def} ::
+## code:: z.do({|x| x.play }) :: || code:: z.do {|x| x.play } ::
+## code:: while({ a < b },{ a = a * 2 }) :: || code:: while { a < b } { a = a * 2 } ::
+## code:: Pfunc({ rrand(3,6) }) :: || code:: Pfunc { rrand(3,6) } ::
 ::
 
-subsection:: calling performList
+note::
+Such trailing function arguments (i.e. outside of parentheses) need be blocks though (Function constructor calls), for both receiver and functional syntax (but not for the binary operator syntax). Functions (previously) assigned to variables or any other expressions that evaluate to functions cannot be used in place of blocks in such shortcuts.
```

Comment:
could you perhaps reword this paragraph so it doesn't have so many parenthetical remarks? they really disrupt the flow.

Reply:
Yeah, that wasn't terribly clear writing on my behalf. I was trying to explain with too many issues simultaneously. What about:

> Trailing arguments must be literal blocks. No other expression may be used as a trailing argument, even if it evaluates to a Function. For example, you cannot use a variable name as a trailing argument, even if this variable was assigned a Function.
>
> Using a selector as an infix binary operator (discussed in the next section) enables a visually similar construct that does allow arbitrary expressions as operands, but these binary-operator constructs technically do not have trailing arguments.

Reply:
sounds good to me, thanks!